File: Lua/LedNumbers.lua
function split(s,delim)
	result = {}
	for match in (s..delim):gmatch("(.-)"..delim) do
		table.insert(result,match)
	end
	return result
end

leds = {[0] = " _  ,| | ,|_| ",
        [1] = "  ,| ,| ",
        [2] = " _  , _| ,|_  ",
        [3] = "_  ,_| ,_| ",
        [4] = "    ,|_| ,  | ",
        [5] = " _  ,|_  , _| ",
        [6] = " _  ,|_  ,|_| ",
        [7] = "_   , |  , |  ",
        [8] = " _  ,|_| ,|_| ",
        [9] = " _  ,|_| , _| "}

io.write("Enter a number: ")
num = io.read()
for i = 1,3 do
	for j = 1, #num do
		line=split(leds[tonumber(string.sub(num,j,j))],",")
		io.write(line[i])
	end
	print("")
end

File: Rust/led_numbers.rs
use std::io;
use std::collections::HashMap;

fn main(){
	let mut leds:HashMap<&str, &str> = HashMap::new();

	leds.insert("0", " _  ,| | ,|_| ");
	leds.insert("1", "  ,| ,| ");
	leds.insert("2", " _  , _| ,|_  ");
	leds.insert("3", "_  ,_| ,_| ");
	leds.insert("4", "    ,|_| ,  | ");	
	leds.insert("5", " _  ,|_  , _| ");
	leds.insert("6", " _  ,|_  ,|_| ");
	leds.insert("7", "_   , |  , |  ");
	leds.insert("8", " _  ,|_| ,|_| ");
	leds.insert("9", " _  ,|_| , _| ");

	println!("Enter a number : ");
	let mut input_text = String::new();
	io::stdin().read_line(&mut input_text)
	           .expect("failed to read");

	let split = input_text.split("");
	let vec: Vec<&str> = split.collect();
	let count = &vec.len() - 2;
	
	for i in 0..3{
		for j in 0..count{
			match leds.get(&vec[j]){
				Some(led_line) => { 
					let line = led_line.split(",");
					let vec_line: Vec<&str> = line.collect();
					print!("{}",&vec_line[i]);
					},
				None => println!("")
			}
		}
		print!("");
	}
	
	println!("");
}
